maestro-studio/web/src/context/ReplContext.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { createContext, ReactNode, useContext, useEffect, useState } from 'react';
+import { Repl, ReplCommand, ReplCommandStatus } from '../helpers/models';
+import { v4 as uuidv4 } from 'uuid';
+import { API } from '../api/api';
+import YAML from 'yaml';
+
+const initialState: Repl = {
+  commands: []
+};
+
+const ReplContext = createContext<{
+  repl: Repl;
+  setRepl: React.Dispatch<React.SetStateAction<Repl>>;
+  errorMessage: string | null;
+  setErrorMessage: React.Dispatch<React.SetStateAction<string | null>>;
+}>({ repl: initialState, setRepl: () => {}, errorMessage: null, setErrorMessage: () => {} });
+
+const restoreRepl = () => {
+  const savedRepl = localStorage.getItem('repl');
+  if (!savedRepl) return initialState
+  return JSON.parse(savedRepl, (key, value) => {
+    if (key === 'command' && !value) return []
+    return value
+  })
+}
+
+export const ReplProvider = ({ children }: {
+  children: ReactNode
+}) => {
+  const [repl, setRepl] = useState<Repl>(() => restoreRepl());
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  useEffect(() => {
+    localStorage.setItem('repl', JSON.stringify(repl));
```

Comment:
@proksh Thoughts on how to best deal with malformatted data in localstorage? Is that necessary?

Reply:
Let's leave as is for now since it's not clear if versioning will be needed long term. We can just remember to add versioning whenever we make a breaking change (and we can treat the absence of a version as version 0)